FxAClient/Frontend/SignIn/FxASignInWebViewController.swift

```diff
@@ -0,0 +1,179 @@
+        let data = [
+            "type": type,
+            "content": content,
+        ]
+        let json = JSON(data).toString(pretty: false)
+        let script = "window.postMessage(\(json), '\(url)');"
+        NSLog("reponse: \(json)")
+        webView!.evaluateJavaScript(script, completionHandler: nil)
+    }
+
+    private func onCanLinkAccount(data: JSON) {
+        //    // We need to confirm a relink - see shouldAllowRelink for more
+        //    let ok = shouldAllowRelink(accountData.email);
+        let ok = true
+        injectData("message", content: ["status": "can_link_account", "data": ["ok": ok]]);
+    }
+
+    // We're not signed in to a Firefox Account at this time, which we signal by returning an error.
+    private func onSessionStatus(data: JSON) {
+        injectData("message", content: ["status": "error"])
+    }
+
+    // We're not signed in to a Firefox Account at this time. We should never get a sign out message!
+    private func onSignOut(data: JSON) {
+        injectData("message", content: ["status": "error"])
+    }
+
+    // The user has signed in to a Firefox Account.  We're done!
+    private func onLogin(data: JSON) {
+        NSLog("Logged in with email: %@ and uid: %@", data["email"].asString!, data["uid"].asString!)
+        injectData("message", content: ["status": "login"])
+        dismissViewControllerAnimated(true, completion: nil)
+        self.delegate?.didSignIn(data)
+    }
+
+    // The content server page is ready to be shown.
+    private func onLoad(data: JSON) {
+        delegate?.didLoad()
+    }
+
+    // Handle a message coming from the content server.
+    func handleRemoteCommand(command: String, data: JSON) {
+        NSLog("command: %@", command)
+        switch command {
+        case "load":
+            onLoad(data)
+        case "login":
+            onLogin(data)
+        case "can_link_account":
+            onCanLinkAccount(data)
+        case "session_status":
+            onSessionStatus(data)
+        case "sign_out":
+            onSignOut(data)
+        default:
+            NSLog("Unexpected remote command received: " + command + ". Ignoring command.");
+        }
+    }
+
+    // Dispatch webkit messages originating from our child webview.
+    func userContentController(userContentController: WKUserContentController, didReceiveScriptMessage message: WKScriptMessage) {
+        if (message.name == "accountsCommandHandler") {
+            let body = JSON(message.body)
+            let detail = body["detail"]
+            handleRemoteCommand(detail["command"].asString!, data: detail["data"])
+        } else {
+            NSLog("Got unrecognized message \(message)")
+        }
+    }
+
+    private func getJS() -> String {
+        let fileRoot = NSBundle.mainBundle().pathForResource("FxASignIn",
+                ofType: "js")
+        return NSString(contentsOfFile: fileRoot!, encoding: NSUTF8StringEncoding, error: nil)!
```

Comment:
I disagree.  We do this in Fennec in many places.  A crash in an early channel (Nightly, Beta) leads to much better user outcomes much faster than trying to deal with a busted build.